src/festim/helpers.py

```diff
@@ -12,16 +15,265 @@ def as_fenics_constant(
         mesh: the mesh of the domiain
 
     Returns:
-        The converted value
+        (fem.Constant) The converted value
 
     Raises:
         TypeError: if the value is not a float, an int or a dolfinx.Constant
     """
     if isinstance(value, (float, int)):
-        return fem.Constant(mesh, dolfinx.default_scalar_type(value))
+        return fem.Constant(mesh, dolfinx.default_scalar_type(float(value)))
     elif isinstance(value, fem.Constant):
         return value
     else:
         raise TypeError(
             f"Value must be a float, an int or a dolfinx.Constant, not {type(value)}"
         )
+
+
+def as_ufl_expression(value, mesh=None, t=None, temperature=None) -> ufl.core.expr.Expr:
+    """Maps a user given function
+
+    Args:
+        value (Callable): the callable to convert
+        mesh (dolfinx.mesh.Mesh): the mesh of the domain
+        t (fem.Constant): the time
+        temperature (fem.Function, fem.Constant or ufl.core.expr.Expr): the temperature
+
+    Returns:
+        (ufl.core.expr.Expr) The mapped function
+    """
+
+    arguments = value.__code__.co_varnames
+
+    kwargs = {}
+    if "t" in arguments:
+        kwargs["t"] = t
+    if "x" in arguments:
+        x = ufl.SpatialCoordinate(mesh)
+        kwargs["x"] = x
+    if "T" in arguments:
+        kwargs["T"] = temperature
+
+    return value(**kwargs)
+
+
+def as_fenics_interpolation_expression(
+    value, function_space, mesh=None, t=None, temperature=None
+) -> fem.Expression:
+    """Converts a callable input value to a fenics expression"""
+
+    mapped_function = as_ufl_expression(
+        value=value, mesh=mesh, t=t, temperature=temperature
+    )
+
+    return fem.Expression(
+        mapped_function,
+        function_space.element.interpolation_points(),
+    )
+
+
+def as_fenics_interp_expr_and_function(
+    value, function_space, mesh=None, t=None, temperature=None
+) -> fem.Function:
+
+    fenics_interpolation_expression = as_fenics_interpolation_expression(
+        value=value,
+        function_space=function_space,
+        mesh=mesh,
+        t=t,
+        temperature=temperature,
+    )
+    fenics_object = fem.Function(function_space)
+    fenics_object.interpolate(fenics_interpolation_expression)
+
+    return fenics_interpolation_expression, fenics_object
```

Comment:
I don't think these functions are very useful. `as_fenics_interp_expr_and_function` basically calls `as_fenics_interpolation_expression` and adds two lines...

`as_fenics_interpolation_expression` calls `as_ufl_expression` and adds one line

Reply:
Oh yeah, meant to merge these two actually, totally forgot